pkg/utilities/components/meta/diff.go

```diff
@@ -340,3 +389,88 @@ func nodesEqual(a, b *yaml.Node, compareComments bool) bool {
 	}
 	return true
 }
+
+func splitManifestFile(combinedYaml []byte) ([]string, [][]byte) {
```

Comment:
WDYT about creating a new helper struct `OrderedMap` that would make it a bit easier to work with this data structure?